pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -667,6 +667,10 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 		ep.HostIdentifier = &endpoint.LbEndpoint_Endpoint{Endpoint: &endpoint.Endpoint{
 			Address: util.BuildInternalAddressWithIdentifier(connectOriginate, net.JoinHostPort(address, strconv.Itoa(port))),
 		}}
+		peerPrincipal, _ := structpb.NewStruct(map[string]any{
```

Comment:
Do we need this for everything doing hbone or just waypoints?

Like does sidecar have the issue today? Seems like probably yes?

Reply:
Yeah I would think so; the simplest implementation seemed to be just always adding the identity in endpoint metadata